### PR TITLE
Extract store and implement migrations support. Migrate to object store with unique ids.

### PR DESCRIPTION
**Pull request by andrey-skl**

### CODE REVIEW

**Comment by andrey-skl** [comment]
@kentcdodds what do you think about this store implementation?

File: src/store.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    get() {
+        const rawData = storage.getItem(this.storeId);
+        return JSON.parse(rawData);
+    }
+
+    applyMigrations(migrations) {
+        const initialData = this.get();
+        const hasMigrationsToApply = migrations.some(m => m.isApplicable(initialData));
```
Comment (kentcdodds) [inline src/store.js]: I'm fine skipping this check and just doing the `reduce` whether needed or not. Sound good?

**Comment by kentcdodds** [comment]
This is great! If you could update the code to remove the `hasMigrationsToApply` check, I'll pull it down and verify it works for me locally. Thanks!

File: src/store.js (hunk 1 above)
Comment (andrey-skl) [inline src/store.js]: @kentcdodds in that way we'll write localStorage every time page opens. This is a real disk write operation which costs something. But for this application we could skip this I guess

Comment (kentcdodds) [inline src/store.js]: Ah! Yeah, probably not necessary. Good point. Alright, let me pull this down and check it out.

**Comment by andrey-skl** [comment]
@kentcdodds I did that, please check this out : )

**Comment by kentcdodds** [comment]
Looks great!

**Comment by andrey-skl** [comment]
@kentcdodds nice, thanks = )

**Pull request merged**